src/main/java/com/poolapp/pool/service/impl/PoolServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package com.poolapp.pool.service.impl;
+
+import com.poolapp.pool.exception.ModelNotFoundException;
+import com.poolapp.pool.model.Pool;
+import com.poolapp.pool.model.PoolSchedule;
+import com.poolapp.pool.repository.PoolRepository;
+import com.poolapp.pool.repository.PoolScheduleRepository;
+import com.poolapp.pool.service.PoolService;
+import com.poolapp.pool.util.ErrorMessages;
+import com.poolapp.pool.mapper.PoolMapper;
+import com.poolapp.pool.mapper.PoolScheduleMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class PoolServiceImpl implements PoolService {
+
+    private final PoolRepository poolRepository;
+    private final PoolScheduleRepository scheduleRepository;
+
+    @Override
+    public Pool createPool(Pool pool) {
+        return poolRepository.save(pool);
+    }
+
+    @Override
+    public Pool getPoolById(Integer id) {
+        return poolRepository.findById(id)
+                .orElseThrow(() -> new ModelNotFoundException(ErrorMessages.POOL_NOT_FOUND + id));
+    }
+
+    @Override
+    public List<Pool> getAllPools(String name) {
```

Comment:
1. getAllPools -> searchPools or getPoolsByFilter
2. we also can search by another fields